### ISSUE TITLE
Ability to Import Submissions

### ISSUE THREAD

**Issue by spiderwisp** [wpninjas/ninja-forms#2235]
Currently you can export submissions, but not import them.

In Ninja Forms 2.0 importing submissions would be a challenge without mapping because field ids would have to match. see https://github.com/wpninjas/ninja-forms/issues/308

However, it's my understanding that Ninja Forms 3.0+ will handle field identification differently and allow field names.

This would be a most useful feature.

**Comment by kstover** [comment]
I think that this could be a cool idea. I think there are some UI questions to be answered, like how do we prompt the user to map the submission data to the form field data. This might be tricky as the submission may or may not have the field label with it. That isn't to say it's impossible, just that we need to talk about it.

**Comment by spiderwisp** [comment]
I had to whip this up for a client. I'm using a modal for the user to map the fields and it looks like this:

![nf-subimport](https://cloud.githubusercontent.com/assets/2864528/21534705/ee07d6ea-cd38-11e6-84e3-24d979e67f20.jpg)

The UI allows you to rearrange the right column to map the fields (needs some refinement for blank labels). 

Then I'm using jQuery CSV to parse the file input and build my mapping array which I send through ajax and then create/update new submissions.

Unfortunately for this particular case this is all done in a custom frontend portal, so my code isn't plug n play. However, if I can manage to get a bit of free time I'll slap it into your plugin.

The php looks a bit like this (removed nonces and validation to simplify for example):

```
$form_id = $_REQUEST['form_id'];
    $form_fields = $_REQUEST['form_fields'];
    $csv_fields = $_REQUEST['csv_fields'];
    $csv_data = $_REQUEST['csv_data'];
    
    $f_data = array_combine(array_intersect_key($form_fields, $csv_fields), array_intersect_key($csv_fields, $form_fields));
    $csv_headers = $csv_data[0];
    
    foreach( $csv_data as $csv_row) {
        $c_data = array_combine(array_intersect_key($csv_headers, $csv_row), array_intersect_key($csv_row, $csv_headers));
        $sub_id = import_submissions_create_submission($form_id);                    
            foreach($f_data as $field_id => $field_label) {
                Ninja_Forms()->sub( $sub_id )->add_field( $field_id, $c_data[$field_label] );
            }
    }
    
    _e( 'Submissions successfully imported', 'vvv' );
    
    exit();
```
And the js looks a bit like this (removed identifying code). Needs more validation and browser checking support for html5 etc.:

     jQuery('body').on( 'click', '#import-submissions', function() {
            
                var form_id = jQuery('#import-submissions-select').val();
                
                var form_fields = [];
                var csv_fields = [];
                
                jQuery('#import-submission-left input[type=text]').each(function(i,v) {
                    form_fields.push(jQuery(this).data('field_id'));
                });
                
                jQuery('#import-submission-right input[type=text]').each(function(i,v) {
                    csv_fields.push(v.value);
                });
                
                var csv_data = null;
                
                var file = document.getElementById('import-submissions-file').files[0];
                
                var reader = new FileReader();
                reader.readAsText(file);
                reader.onload = function(event) {
                    
                    var csvData = event.target.result;
                    csv_data = jQuery.csv.toArrays(csvData);
                    
                    if (csv_data && csv_data.length > 0) {
                        
                      jQuery.ajax({  
                        type: 'POST',  
                        url: '<?php echo admin_url( 'admin-ajax.php' ) ?>',  
                        data: { 
                            action : 'import_submissions',
                            form_id: form_id,
                            form_fields: form_fields,
                            csv_fields: csv_fields,
                            csv_data: csv_data
                        },
                        success: function(textStatus){ 
                            jQuery( '#portal-message' ).stop(true,true).fadeIn();
                            jQuery( '#portal-message' ).html(csv_data.length + ' submissions successfully imported').delay(2000).fadeOut("slow");                      
                        },  
                        error: function(MLHttpRequest, textStatus, errorThrown){  
                            alert(errorThrown);  
                        }  
                      });
                      
                      
                    } else {
                        alert('File seems to have no valid data!');
                    }
                };
                
                
            });`

**Comment by jameslaws** [comment]
This looks pretty decent. Could you submit that as PR for further development.

**Comment by kstover** [comment]
@kjohnson Can you take a look at this?

**Comment by j4jes** [comment]
I would buy this feature if available.

**Comment by AdnanMuhib** [comment]
is this feature available in v3.0?
I don't see any options but I need that feature to work.

**Comment by marounmelhem** [comment]
I've tried exporting/importing using the default WordPress exporter/importer (located under tools) and it worked great! All forms submissions are there along with the form fields. (Make sure to select "submissions" type when exporting)  
You just need to make sure the form IDs match between the 2 sites. Note that exporting and importing forms using the NinjaForms forms export tool could generate different form IDs.

Ref: [How to use the WordPress importer](https://www.hostinger.com/tutorials/wordpress-importer/)

**Comment by JoseAMijaresRS** [comment]
> 
> 
> I've tried exporting/importing using the default WordPress exporter/importer (located under tools) and it worked great! All forms submissions are there along with the form fields. (Make sure to select "submissions" type when exporting)
> You just need to make sure the form IDs match between the 2 sites. Note that exporting and importing forms using the NinjaForms forms export tool could generate different form IDs.
> 
> Ref on [How to use the WordPress importer](https://www.hostinger.com/tutorials/wordpress-importer/)

**Maroun**, sorry to bother you, could you please shrtly explain how did you convert the submission contacts excel xls file to wordpress import accepted WXR format? 
I tried saving the xls to xml but in the import I got this:
"This does not appears to be a WXR file, the WXR version number is missing or invalid"
Thank you
Jose
